Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -2817,9 +2817,16 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
     }
 
     case ToNumber: {
+        if (node->child1().useKind() == StringUse) {
+            setNonCellTypeForNode(node, SpecBytecodeNumber);
+            break;
+        }
+
         JSValue childConst = forNode(node->child1()).value();
+
         if (childConst && childConst.isNumber()) {
-            didFoldClobberWorld();
+            if (node->child1().useKind() != StringUse)
```

Comment:
Isn't this always true?  You already filtered out the `node->child1().useKind() == StringUse` case at line 2820 above.

Reply:
Yeah, we should remove this.